drivers/serial/uart_nrfx_uart.c

```diff
@@ -467,13 +494,49 @@ static const struct uart_driver_api uart_nrfx_uart_driver_api = {
 #endif /* CONFIG_UART_0_INTERRUPT_DRIVEN */
 };
 
-DEVICE_AND_API_INIT(uart_nrfx_uart0,
-		    CONFIG_UART_0_NAME,
-		    uart_nrfx_init,
-		    NULL,
-		    NULL,
-		    /* Initialize UART device before UART console. */
-		    PRE_KERNEL_1,
-		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &uart_nrfx_uart_driver_api);
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static void uart_nrfx_set_power_state(u32_t new_state)
+{
+	if (new_state == DEVICE_PM_ACTIVE_STATE) {
+		nrf_uart_enable(uart0_addr);
+		nrf_uart_task_trigger(uart0_addr, NRF_UART_TASK_STARTRX);
+	} else {
+		assert(new_state == DEVICE_PM_LOW_POWER_STATE ||
+		       new_state == DEVICE_PM_SUSPEND_STATE ||
+		       new_state == DEVICE_PM_OFF_STATE);
+		nrf_uart_disable(uart0_addr);
```

Comment:
What happens to Rx (3-wire, 5-wire) in power save states?

Reply:
I am not sure what do you mean by "3-wire" and "5-wire" (without and with flow control?), but anyway, both RX and TX are disabled in all power save states, so nothing can be received or transmitted when the UART device is instructed to save power. That's what the PR is about. It provides a way of disabling RX in UART using Zephyr APIs (you can't do it with the current UART API). But it's up to the application when the UART will be disabled. It must explicitly call `device_set_power_state` (and point the UART device and the requested power state) from its `_sys_soc_suspend` hook.

Reply:
I meant applications designed to receive data over UART without h/w flowcontrol should not disable UART, else will loose the data. Also, application with h/w flowcontrol will need a way to wakeup-on-uart implementation. This PR changes makes sense for Tx only applications; and the PR/commit title and implementation should be restricted to Tx only power management.

Reply:
Applications expecting that something may appear on RX will just not set any power save state for the UART. There is no point in doing it only for TX part, since the transmitter is activated only for the time when something is being sent.
This PR does not force anyone to put the UART into a power saving mode (i.e. to disable it), it only creates such possibility for the ones needing this.

Reply:
Ok, I am convinced with the fact that applications designed to Rx will not implement _sys_soc_suspend. (but thinking aloud, may be other peripherals could be put in low power; but then we need multiple power profile mode, not the single power save state).